File: Tests/EdgeCasesAnnotations/Output/member-types.ts
/* eslint-disable */
// tslint:disable

// Covers the member level annotations: renaming a member, renaming via replace and overriding the
// generated type.
export class MemberTypes {
    public renamedField: string = "";
    public untouchedProperty: string = "";
    public renamedProperty: string = "";
    public niceNameProperty: string = "";
    public typeNameOverrideProperty: string | undefined;

    public constructor(init?: Partial<MemberTypes>) {
        Object.assign(this, init);
    }
}

// outputid:0f0bd27f-b1e8-4ba0-bb6e-c0f7dfef979b

File: Tests/EdgeCasesAnnotations/MemberTypes.cs
using KY.Generator;
#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member

namespace EdgeCasesAnnotations;

/// <summary>
/// Covers the member level annotations: renaming a member, renaming via replace and overriding the
/// generated type.
/// </summary>
[GenerateTypeScriptModel("Output")]
public class MemberTypes
{
    public string UntouchedProperty { get; set; } = "";

    [GenerateProperty(Name = "renamedProperty")]
    public string OriginalNameProperty { get; set; } = "";

    [GenerateProperty(Replace = "Ugly", With = "Nice")]
    public string UglyNameProperty { get; set; } = "";

    [GenerateProperty(TypeName = "string")]
    public int TypeNameOverrideProperty { get; set; }

    [GenerateField(Name = "renamedField")]
    public string OriginalNameField = "";
}
